website/docs/developer/research/research-pushing-docker-image.md

```diff
@@ -0,0 +1,117 @@
+---
+id: research-pushing-docker-image
+title: Notes on pushing and deploying docker images.
+description: Research notes on pushing Docker images for end users
+---
+
+## Deploying ECR
+
+ECR repo can be deploying using terraform as shown below.
+
+```terraform
+resource "aws_ecr_repository" "prototype_repo" {
+  name                 = "prototype_repo"
+
+  image_scanning_configuration {
+    scan_on_push = false
+  }
+}
+```
+
+## Github Packages
+
+[Github packages](https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages) looks like a good platform to store public or private docker images. GitHub Packages usage is free for public packages. For private packages, each account on GitHub receives a certain amount of free storage and data transfer, depending on the product used with the account. Any usage beyond the included amounts is controlled by spending limits. Currently, this github package feature is already being used by our NASA [repository](https://github.com/orgs/nasa/packages).
+
+The steps to push a docker image to github repository are shown below. Make sure Docker CLI is installed on your machine.
+
+- Create a DockerFile that will create the image. 
+
+```yaml
+#This is a sample image
+FROM alpine:3.4
+RUN apk update
+RUN apk add vim 
+CMD [“echo”,”Sample image created”]
+```
+- Build the image using `docker build -t sample-image .`. 
+- Once the image is there, the next step is to push to github. However, the user will first need to authenticate with github using a `personal access token`. See details [here](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) on how to create the token. 
+:::important
+Github packages might need to be approved by NASA admins for use. In addition, proper permissions need to be granted to the token such as `write_package` in order to push to github.
+:::
+
+- Once the token is created, sign in to the github container registry using:
+```bash
+export CR_PAT=YOUR_PERSONAL_TOKEN
+echo $CR_PAT | docker login ghcr.io -u <GITHUB_USERNAME> --password-stdin
+```
+Replace `<USERNAME>` with your github username. It will show `Login Succeeded` upon success.
+
+- Tag the docker image using the following format. Note that the format must be `ghcr.io/<GITHUB_USERNAME>/<image_name>`
```

Comment:
Since cumulus-orca is under `nasa`, the name should be `ghcr.io/nasa/<image_name>:<image_tag>`

As an example, see https://github.com/nasa/astrobee/pkgs/container/astrobee

Reply:
Image name should be cumulus-orca/<image_name>:tag in case / are accepted
unless / are accepted, it will be cumulus-orca-<image_name>:tag

tag will always be the version

Reply:
updated to include image naming convention